multiversx/mx-chain-go
# Shard ID in all outport structures

## Reasoning behind the pull request
- Added the shard ID in all the outport structures
  
## Proposed changes
- Changed the version of the `mx-chain-core` module (https://github.com/multiversx/mx-chain-core-go/pull/244)
- Populated `shardID` field in the outport structures
- Changed the version of the `mx-chain-es-indexer-go` module ( https://github.com/multiversx/mx-chain-es-indexer-go/pull/236)

## Testing procedure
- Verify if the explorer is working correctly 
- Verify if the log `WARN [2023-07-20 10:21:15.370]  [2/1084/15621812/(START_ROUND)] dataDispatcher.doWork could not index item (will retry) error =....` will no longer appears

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?
